src/python_testing/TC_DeviceConformance.py

```diff
@@ -236,6 +238,51 @@ def test_TC_IDM_10_3(self):
         if not success:
             self.fail_current_test("Problems with cluster revision on at least one cluster")
 
+    def test_IDM_10_5(self):
+        success = True
+        warn_on_extra_clusters = self.user_params.get("warn_on_extra_clusters", True)
+        test_name = self.get_test_name()
+
+        for endpoint_id, endpoint in self.endpoints.items():
+            device_types = [x for x in endpoint[Clusters.Descriptor]
+                            [Clusters.Descriptor.Attributes.DeviceTypeList] if x.deviceType <= 0xBFFF]
+            endpoint_clusters = []
+            for device_type in device_types:
+                device_type_id = device_type.deviceType
+                if device_type_id not in self.xml_device_types.keys():
+                    # TODO: report a problem here
+                    continue
+                # TODO: check revision
+                xml_device = self.xml_device_types[device_type_id]
+                server_clusters = [x for x in endpoint[Clusters.Descriptor]
+                                   [Clusters.Descriptor.Attributes.ServerList] if x <= 0x7FFF]
```

Comment:
I would add a named constant for 0x7FFFF and any other constants as well.

Reply:
I seem to remember that we already have 0x7FFF defined in some of the other IDM tests 🤔